CMakeLists.txt

```diff
@@ -79,13 +80,25 @@ if(NOT USE_HIP_CPU)
       set(DEFAULT_AMDGPU_TARGETS "gfx803;gfx900;gfx906;gfx908;gfx1030")
     endif()
   endif()
-  set(AMDGPU_TARGETS "${DEFAULT_AMDGPU_TARGETS}" CACHE STRING "List of specific machine types for library to target")
-  set(AMDGPU_TEST_TARGETS "" CACHE STRING "List of specific device types to test for") # Leave empty for default system device
+  if (AMDGPU_TARGETS)
+    if( AMDGPU_TARGETS STREQUAL "all" )
+      set( gpus "${DEFAULT_AMDGPU_TARGETS}")
+    else()
+      set( gpus "${AMDGPU_TARGETS}")
+    endif()
+    # must FORCE set this AMDGPU_TARGETS before any find_package( hip ...), in this file
+    # to override CACHE var and set --offload-arch flags via hip-config.cmake hip::device dependency
+    set( AMDGPU_TARGETS "${gpus}" CACHE STRING "AMD GPU targets to compile for" FORCE )
+  endif()
+
+  #set(AMDGPU_TARGETS "${DEFAULT_AMDGPU_TARGETS}" CACHE STRING "List of specific machine types for library to target" FORCE)
+  set(AMDGPU_TEST_TARGETS "" CACHE STRING "List of specific device types to test for" FORCE) # Leave empty for default system device
```

Comment:
This change causes some issue for our CI. The ```AMDGPU_TEST_TARGETS``` only usef for the test building and  the only purpose to change the generated test names:

```
foreach(AMDGPU_TARGET IN LISTS AMDGPU_TEST_TARGETS)
      add_test("${AMDGPU_TARGET}-${TEST_NAME}" ${TEST_TARGET})
      set_tests_properties("${AMDGPU_TARGET}-${TEST_NAME}"
          PROPERTIES
              RESOURCE_GROUPS "1,${AMDGPU_TARGET}:1"
              LABELS "hip;${AMDGPU_TARGET}"
      )
    endforeach()
 ```
 
 Can we get rid of it or move it to the test CMakeLists.txt and avoid the ```FORCE``` parameter.

Reply:
Done.  Moved set(AMDGPU_TEST_TARGETS "" CACHE STRING "List of specific device types to test for" FORCE) to test/rocprim/CMakeLists.txt and removed the FORCE parameter.  Fixed in upcoming update to this PR.